Fixes format specification warning when compiling 32-bit
code on Darwin. SInt32 is int on 64-bit systems, long
on 32-bit systems.

diff --git a/src/bonjour.c b/src/bonjour.c
index ac710fda8..2e7df1a0c 100644
--- a/src/bonjour.c
+++ b/src/bonjour.c
@@ -19,6 +19,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <inttypes.h>
 
 #include "tvheadend.h"
 #include "bonjour.h"
@@ -37,8 +38,8 @@ static void
 bonjour_callback(CFNetServiceRef theService, CFStreamError* error, void* info)
 {  
   if (error->error) {
-    tvherror(LS_BONJOUR, "callback error (domain = %ld, error =%d)",
-             error->domain, error->error);   
+    tvherror(LS_BONJOUR, "callback error (domain = %ld, error =%"PRId32")",
+             error->domain, (int32_t) error->error);
   } 
 }
 
@@ -85,7 +86,8 @@ bonjour_start_service(CFNetServiceRef *svc, const char *service_type,
 
   if (!CFNetServiceRegisterWithOptions(*svc, 0, &error))
     tvherror(LS_BONJOUR, "registration failed (service type = %s, "
-             "domain = %ld, error =%d)", service_type, error.domain, error.error); 
+             "domain = %ld, error =%"PRId32")", service_type, error.domain,
+             (int32_t) error.error);
   else
     tvherror(LS_BONJOUR, "service '%s' successfully established",
              service_type);
